### PR TITLE
✨Add version to user-agent for AWS API calls

### LINKED ISSUE #1242: Set user-agent for calls to AWS
/kind feature

**Describe the solution you'd like**
For debugging, I go through CloudTrail logs in Athena, e.g.

``` sql
SELECT eventtime,
         eventsource,
         eventname,
         useragent
FROM "cloudtrail"."cloudtrail"
WHERE awsregion='us-east-1' limit 1000
```

eventtime | eventsource | eventname | useragent
-- | -- | -- | --
2019-10-22T13:42:17Z | ec2.amazonaws.com | DescribeInstances | kubernetes/v1.15.3 aws-sdk-go/1.16.26 (go1.12.9; linux; amd64)
2019-10-22T12:34:16Z | elasticloadbalancing.amazonaws.com | RegisterInstancesWithLoadBalancer | aws-sdk-go/1.20.19 (go1.12.9; linux; amd64)

Core Kubenetes components identify themselves as
`kubernetes/v1.15.3 aws-sdk-go/1.16.26 (go1.12.9; linux; amd64)`

but CAPA uses the default 
`aws-sdk-go/1.20.19 (go1.12.9; linux; amd64)`

A sample UA could be:

`aws.cluster.x-k8s.io/0.5.0 aws-sdk-go/1.20.19 (go1.12.9; linux; amd64)`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 0.4.3
- Kubernetes version: (use `kubectl version`):  N/A
- OS (e.g. from `/etc/os-release`): N/A

### PR DESCRIPTION
**Pull request by aaroniscode**
**What this PR does / why we need it**:
The AWS SDK adds the SDK version to the user-agent of API calls. This adds the version of the Cluster API AWS provider to the user-agent string.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1242

### CODE REVIEW

**Comment by aaroniscode** [comment]
For the reviewers -- note that because version information was needed for the controller (and not just `clusterawsadm` I refactored the version code to be a shared library for both to use.

I modeled the changes to the `clusterawsadm` version command to be very similar to how `kubectl` and `kubeadm` work.

File: Dockerfile [hunk 1]
```diff
@@ -36,8 +36,9 @@ RUN wget --output-document /restart.sh --quiet https://raw.githubusercontent.com
 
 # Build
 ARG ARCH
+ARG LDFLAGS
```
Comment (vincepri) [inline Dockerfile]: Shouldn't this have a default? Would this affect tilt work? 

/cc @ncdc

Comment (ncdc) [reply Dockerfile]: Doesn't affect tilt. My tilt setup uses a separate dockerfile just for tilt

Comment (aaroniscode) [reply Dockerfile]: Hmm... is the Dockerfile used anywhere outside of `make docker-build`?

The Makefile uses a `--build-arg` to set LDFLAGS, just like ARCH which doesn't have a default.

It's not clear to me that it needs a default, but I still feel like I'm learning the code base. When testing builds I had errors in the `version.sh` script and it just built and the version variables were empty.  Also just tested a build with empty LDFLAGS and it built without error. Of course it wouldn't have version info.

Let me know what the team thinks and if we should have a default, what you would like set and I'm happy to update!

Comment (randomvariable) [reply Dockerfile]: I had previously tested this prior to the PR. These build fine on BuildKit, and the flags are translated to empty strings when they're not present.

Comment (randomvariable) [reply Dockerfile]:
```
➜ docker build .
[+] Building 53.2s (16/16) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                            0.0s
 => => transferring dockerfile: 1.89kB                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                               0.0s
 => => transferring context: 171B                                                                                                                               0.0s
 => [internal] load metadata for gcr.io/distroless/static:latest                                                                                                0.4s
 => [internal] load metadata for docker.io/library/golang:1.12.10                                                                                               0.9s
 => [stage-1 1/3] FROM gcr.io/distroless/static:latest@sha256:c6d5981545ce1406d33e61434c61e9452dad93ecd8397c41e89036ef977a88f4                                  1.2s
 => => resolve gcr.io/distroless/static:latest@sha256:c6d5981545ce1406d33e61434c61e9452dad93ecd8397c41e89036ef977a88f4                                          0.0s
 => => sha256:c6d5981545ce1406d33e61434c61e9452dad93ecd8397c41e89036ef977a88f4 526B / 526B                                                                      0.0s
 => => sha256:e4d2a899b1bf57a286ca955c1f4a6b6fbb5963292e334252b4de6e8563eebee1 434B / 434B                                                                      0.0s
 => => sha256:9ff2acc3204b4093126adab3fed72de8f7bbfe332255b199c30b8b185fcf6923 655.32kB / 655.32kB                                                              0.6s
 => => extracting sha256:9ff2acc3204b4093126adab3fed72de8f7bbfe332255b199c30b8b185fcf6923                                                                       0.1s
 => [builder 1/8] FROM docker.io/library/golang:1.12.10@sha256:c2f4fdb24e20b6fdd8f157bbc42535d3ed85e397fd40e964816319b3bb277947                                 0.0s
 => [internal] load build context                                                                                                                               0.1s
 => => transferring context: 1.93MB                                                                                                                             0.0s
 => CACHED [builder 2/8] WORKDIR /workspace                                                                                                                     0.0s
 => [builder 3/8] COPY go.mod go.mod                                                                                                                            0.1s
 => [builder 4/8] COPY go.sum go.sum                                                                                                                            0.0s
 => [builder 5/8] RUN go mod download                                                                                                                          25.5s
 => [builder 6/8] COPY ./ ./                                                                                                                                    0.1s
 => [builder 7/8] RUN wget --output-document /restart.sh --quiet https://raw.githubusercontent.com/windmilleng/rerun-process-wrapper/master/restart.sh  &&      1.0s
 => [builder 8/8] RUN CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH}     go build -a -ldflags "${LDFLAGS} -extldflags '-static'"     -o manager .                     24.8s
 => [stage-1 2/3] COPY --from=builder /workspace/manager .                                                                                                      0.1s
 => exporting to image                                                                                                                                          0.3s
 => => exporting layers                                                                                                                                         0.3s
 => => writing image sha256:8d2416ef793d77486de21ffd6e9c5bd93898ce0c65bc6e24cfd7862d553642d1                                                                    0.0s
```

and

```
➜ go run ./cmd/clusterawsadm version
clusterawsadm version: &version.Info{Major:"", Minor:"", GitVersion:"", GitCommit:"", GitTreeState:"", BuildDate:"", GoVersion:"go1.13.1", AwsSdkVersion:"v1.25.16", Compiler:"gc", Platform:"linux/amd64"}
```

File: Makefile [hunk 2]
```diff
@@ -96,7 +96,7 @@ binaries: manager clusterawsadm ## Builds and installs all binaries
```
Comment (vincepri) [inline Makefile]: `LDFLAGS` variable is defined on line 236, can we move it before the `Help` section now that it's being used in multiple places?

Comment (aaroniscode) [reply Makefile]: @vincepri moved `LDFLAGS` up in the Makefile before the `Help` section.
